core/src/main/scala/org/apache/spark/rdd/JdbcRDD.scala

```diff
@@ -70,7 +71,8 @@ class JdbcRDD[T: ClassTag](
     }).toArray
   }
 
-  override def compute(thePart: Partition, context: TaskContext) = new NextIterator[T] {
+  override def compute(thePart: Partition, context: TaskContext): Iterator[T] = new NextIterator[T]
+  {
```

Comment:
I tried this one, but it looks slightly weirder if we wrap. I'm leaving it as is, but fixed the other ones.